app/src/pages/[resourceType]/[id].tsx

```diff
@@ -28,6 +31,45 @@ export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType
     /* eslint-enable @typescript-eslint/no-explicit-any */
   }, []);
 
+  const [loadingIconVisible, setLoadingIconVisible] = useState(false);
+  const { data: dataRequirements, refetch } = trpc.service.getDataRequirements.useQuery(
+    { resourceType: jsonData.resourceType, id: jsonData.id as string },
+    { enabled: false }
+  );
+  //useEffect checks if the data Requirements were loaded and whether the button was clicked
+  useEffect(() => {
+    if (dataRequirements && loadingIconVisible) {
+      setTimeout(() => {
+        if (dataRequirements?.Library?.resourceType == 'Library') {
+          notifications.show({
+            id: 'requirements',
+            withCloseButton: true,
+            autoClose: 2000,
+            title: 'Successful Fetch',
+            message: 'Data requirements successfully fetched',
+            color: 'teal',
+            style: { backgroundColor: 'white' },
+            icon: <CircleCheck />,
+            loading: false
+          });
+        } else {
+          notifications.show({
+            id: 'no-requirements',
+            withCloseButton: true,
+            autoClose: 3000,
+            title: 'No Data Requirements Found',
+            message: 'No data requirements were found for this resource',
+            color: 'red',
+            style: { backgroundColor: 'white' },
+            icon: <AbacusOff />,
+            loading: false
+          });
```

Comment:
I also think we may have to change some things around with error handling if we want to utilize `onSuccess`, `onError`, and `isLoading`. I think we should utilize trpc error handling within the `getDataRequirements` procedure so that we can actually use these properties. Right now, we cannot use them because the fetch is always successful even if it is an Operation outcome, it is still returning something. Maybe we can make these changes in the procedure in `service.ts` by utilizing [trpc errors](https://trpc.io/docs/server/error-handling). Basically we can check what the details.text is and throw a TRPC error based on what kind of error it is. I haven't actually tried this, but I think it'll work.

Reply:
Just implemented your second suggestion Sarah!So now if that same error were to pop up it should now show you the `"text"` portion of the resource. Thank you both for your great suggestions!

Reply:
![Screenshot 2023-06-20 at 12 32 16 PM](https://github.com/projecttacoma/measure-repository/assets/87094479/c819a9c8-8b0a-4e44-ad2e-48f3b8e5b47d)
Woohoo! This is a great change!